crates/polars-core/src/frame/asof_join/asof.rs

```diff
@@ -182,16 +182,102 @@ pub(super) fn join_asof_backward<T: PartialOrd + Copy + Debug>(
     out
 }
 
+pub(super) fn join_asof_nearest_with_tolerance<
+    T: PartialOrd + Copy + Debug + Sub<Output = T> + Add<Output = T> + Bounded,
+>(
+    left: &[T],
+    right: &[T],
+    tolerance: T,
+) -> Vec<Option<IdxSize>> {
+    let n_left = left.len();
+    let mut out = Vec::with_capacity(n_left);
```

Comment:
Can we allocate after `if left.is_empty()`. That saves going through the allocator.

Reply:
Sure thing. I was also wondering if the emptiness of the left/right vectors was checked earlier on (it probably should be if it's not). I'll assume no for now.

Reply:
@ritchie46 wait, don't we still have to return an empty vec? What do I return if `left.is_empty()` then?

Reply:
Oh, just a `Vec::new()`.